src/Nncase.Tests/Core/UnitTestTensorOfT.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Copyright (c) Canaan Inc. All rights reserved.
+// Licensed under the Apache license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Nncase;
+using Xunit;
+
+namespace Nncase.Tests.CoreTest;
+
+public sealed class UnitTestTensorOfT
+{
+    public static unsafe IEnumerable<object[]> TestSetValueData =>
+        new[]
+        {
+            new object[] { 0, 100 },
+            new object[] { 1, 200 },
+            new object[] { 3, 300 },
+            new object[] { 4, 400 },
+            new object[] { 5, 500 },
+            new object[] { 6, 600 },
+            new object[] { 7, 700 },
+        };
+
+    [Fact]
+    public void TestICollection()
+    {
+        var array1 = new int[] { 1, 2, 3, 4, 5, 6, 7, 8 };
+        var momory = new Memory<int>(array1);
+        var t = new Tensor<int>(momory, new int[] { 1, 1, 2, 4 });
+        ICollection<int> c = t;
+        Assert.Equal(array1.Length, c.Count);
+        Assert.False(c.IsReadOnly);
+
+        Assert.Throws<InvalidOperationException>(() => c.Add(100));
+        Assert.Throws<InvalidOperationException>(() => c.Remove(8));
+
+        var array2 = new int[array1.Length];
+        c.CopyTo(array2, 0);
+        Assert.True(Enumerable.SequenceEqual(array1, array2));
+
+        c.Clear();
+
+        Assert.Equal(array1.Length, c.Count);
+        for (int i = 0; i < array1.Length; i++)
+        {
+            Assert.Equal(0, t.GetValue(i));
+        }
+    }
+
+    [Fact]
+    public void TestIReadOnlyCollection()
```

Comment:
should test default value when only set length, because it may be initialized to the wrong value